app/models/article.rb

```diff
@@ -112,7 +112,7 @@ class Article < ApplicationRecord
 
   scope :admin_published_with, lambda { |tag_name|
     published
-      .where(user_id: SiteConfig.staff_user_id)
+      .where(user_id: User.with_role(:super_admin).pluck(:id) + User.with_role(:admin).pluck(:id))
```

Comment:
Is there a more efficient query we could use here?

Reply:
This will result in a single query:

```suggestion
      .where(user_id: User.with_role(:super_admin).union(User.with_role(:admin)).select(:id))
```

according to Blazer, with this change, this should take around 210 ms.

*BUT*

with separate queries it should take less than 10 ms (I ran `EXPLAIN ANALYZE` on each separate step and added them together). 

That's because it can use indexes it's not using with the subquery. So I think your option is the most efficient one even if it results in 3 queries